packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -295,11 +322,21 @@ class Plugin {
     return errors;
   }
 
+  static bool _providesDefaultPackageForPlatform(YamlMap platformsYaml, String platformKey) {
+    if (!platformsYaml.containsKey(platformKey)) {
+      return false;
+    }
+    if ((platformsYaml[platformKey] as YamlMap).containsKey(kDefaultPackage)) {
```

Comment:
Should this check that `platformsYaml[platformKey]` is a `YamlMap` first before doing the `as`?  Is that checked in a validate step somewhere else?

Reply:
Done!